pkg/sql/jobs/registry.go

```diff
@@ -289,7 +289,9 @@ func (r *Registry) maybeAdoptJob(ctx context.Context, nl nodeLiveness) error {
 			continue
 		}
 
-		go func() {
+		ctx := stopper.WithCancel(ctx)
```

Comment:
This looks like a good practice, but it isn't used anywhere else in the entire code base, so I'm honestly not sure about it. I'd think that we should already be cancelling the top-level server context passed to `(*Registry).Start` in the first place, but it actually doesn't look like we do.

@tschottdorf, since it looks like you wrote `(*Stopper).WithCancel`, do you remember when it's necessary?